modules/serverpod_auth/serverpod_auth_server/lib/src/business/user_authentication.dart

```diff
@@ -34,21 +45,73 @@ class UserAuthentication {
       method: method,
     );
 
-    session.updateAuthenticated(AuthenticationInfo(userId, scopes));
     var result = await AuthKey.db.insertRow(session, authKey);
+
+    if (updateSession) {
+      session.updateAuthenticated(
+        AuthenticationInfo(
+          userId,
+          scopes,
+          authId: '${result.id}',
+        ),
+      );
+    }
     return result.copyWith(key: key);
   }
 
-  /// Signs out a user from the server and deletes all authentication keys.
-  /// This means that the user will be signed out from all connected devices.
-  static Future<void> signOutUser(Session session, {int? userId}) async {
+  /// Signs out the user from all devices.
+  static Future<void> signOutUser(
+    Session session, {
+    int? userId,
+  }) async {
     userId ??= (await session.authenticated)?.userId;
     if (userId == null) return;
 
-    await session.db
-        .deleteWhere<AuthKey>(where: AuthKey.t.userId.equals(userId));
-    await session.messages
-        .authenticationRevoked(userId, RevokedAuthenticationUser());
-    session.updateAuthenticated(null);
+    // Delete all authentication keys for the user
+    var auths = await AuthKey.db.deleteWhere(
+      session,
+      where: (row) => row.userId.equals(userId),
+    );
+
+    if (auths.isEmpty) return;
+
+    // Notify clients about the revoked authentication for the user
+    await session.messages.authenticationRevoked(
+      userId,
+      RevokedAuthenticationUser(),
+    );
+
+    // Clear session authentication if the signed-out user is the currently authenticated user
+    var authInfo = await session.authenticated;
+    if (userId == authInfo?.userId) {
+      session.updateAuthenticated(null);
+    }
+  }
+
+  /// Signs out the user from the current device.
+  static Future<void> revokeAuthKey(
+    Session session, {
+    required int authKeyId,
```

Comment:
I am not sure if we should do this, this is business layer, someone can use it to directly pass a key, and it will be confused for them if we do this, what do you think ?

Reply:
I understand your perspective, in this business layer all authKeyIds are ints. But if we don't take in the string that is stored in the `AuthenticationInfo` the user of the interface needs to understand what we are storing there and how it is converted. I think that implementation details would cause a lot more digging. I think it might just be simpler to do the conversion on the inside.

Reply:
okay, thank you